test/e2e/framework/job/fixtures.go

```diff
@@ -92,6 +92,11 @@ func NewTestJobOnNode(behavior, name string, rPol v1.RestartPolicy, parallelism,
 		}
 	}
 	switch behavior {
+	case "neverTerminate":
+		// this job is being used in an upgrade job see test/e2e/upgrades/apps/job.go
+		// it should never be optimized, as it always has to restart during an upgrade
+		// and continue running
+		job.Spec.Template.Spec.Containers[0].Command = []string{"sleep", "1000000"}
```

Comment:
non-blocking nit: we could have `TerminationGracePeriodSeconds: 1` to make it faster :)

Reply:
Great idea, updated.

Reply:
You probably need `= ptr.To[int64](1)`